app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,544 @@
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.LocationSettingsStates;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.tasks.Task;
+import dagger.android.AndroidInjection;
+import java.io.File;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function0;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.base.BaseActivity;
+import org.kiwix.kiwixmobile.main.MainActivity;
+import org.kiwix.kiwixmobile.utils.AlertDialogShower;
+import org.kiwix.kiwixmobile.utils.KiwixDialog;
+import org.kiwix.kiwixmobile.wifi_hotspot.HotspotService;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.SelectionMode;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BookOnDiskDelegate;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BooksOnDiskAdapter;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BooksOnDiskListItem;
+
+import static org.kiwix.kiwixmobile.utils.StyleUtils.dialogStyle;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.getAddress;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.isServerStarted;
+
+public class ZimHostActivity extends BaseActivity implements
+    ServerStateListener, ZimHostContract.View {
+
+  @BindView(R.id.startServerButton)
+  Button startServerButton;
+  @BindView(R.id.server_textView)
+  TextView serverTextView;
+
+  @Inject
+  ZimHostContract.Presenter presenter;
+
+  @Inject AlertDialogShower alertDialogShower;
+
+  public static final String ACTION_TURN_ON_AFTER_O = "Turn_on_hotspot_after_oreo";
+  public static final String ACTION_TURN_OFF_AFTER_O = "Turn_off_hotspot_after_oreo";
+  public static final String ACTION_IS_HOTSPOT_ENABLED = "Is_hotspot_enabled";
+  public static final String ACTION_START_SERVER = "start_server";
+  public static final String ACTION_STOP_SERVER = "stop_server";
+  public static final String SELECTED_ZIM_PATHS_KEY = "selected_zim_paths";
+  private static final String IP_STATE_KEY = "ip_state_key";
+  private static final String TAG = "ZimHostActivity";
+  private static final int MY_PERMISSIONS_ACCESS_FINE_LOCATION = 102;
+  private static final int LOCATION_SETTINGS_PERMISSION_RESULT = 101;
+  private Intent serviceIntent;
+  private Task<LocationSettingsResponse> task;
+
+  private BooksOnDiskAdapter booksAdapter;
+  HotspotService hotspotService;
+  private String ip;
+  boolean bound;
+  private ServiceConnection serviceConnection;
+  private ArrayList<String> selectedBooksPath = new ArrayList<>();
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_zim_host);
+
+    AndroidInjection.inject(this);
```

Comment:
when you extend baseActivity you do not need to call this

Reply:
Fixed. fc4ba04544a69ac85388a2c70642ea51f6b13d14